src/android/FirebaseConfigPlugin.java

```diff
@@ -95,6 +95,23 @@ protected void getString(String key, CallbackContext callbackContext) {
         callbackContext.success(getValue(key).asString());
     }
 
+    @CordovaMethod
+    protected void getValueSource(String key, CallbackContext callbackContext) {
+        switch (getValue(key).getSource()) {
+            case FirebaseRemoteConfig.VALUE_SOURCE_REMOTE:
+                callbackContext.success("remote");
```

Comment:
Instead of strings you can define static constants with appropriate integer values:
```java
@CordovaMethod
protected void getValueSource(String key, CallbackContext callbackContext) {
    callbackContext.success(getValue(key).getSource());
}
```
On JS side:
```js
module.exports = {
    VALUE_SOURCE_STATIC: 0,
    VALUE_SOURCE_DEFAULT: 1,
    VALUE_SOURCE_REMOTE: 2,
    ...
}
```
In result you don't need to hardcode string values in app code:
```js
remoteConfig.getValueSource("myKey").then((source) => {
    if (source === remoteConfig.VALUE_SOURCE_DEFAULT) {
        ...
    }
})
```

Note that the same constants will be used for iOS so you might need additional logic there.

Reply:
Example of project where similar approach was implemented https://github.com/chemerisuk/cordova-plugin-idfa#tracking-permission-values

Reply:
I implemented what you've suggested, please let me know if it works and then I'll update the doc.

Reply:
Although I wasn't able to test it in my app because it requires some rewriting, I'll do in couple of days.